sponge/src/main/java/com/vexsoftware/votifier/sponge/VotifierPlugin.java

```diff
@@ -366,15 +359,9 @@ public void onForward(final Vote v) {
         if (debug) {
             logger.info("Got a forwarded vote -> " + v);
         }
-        Sponge.getScheduler().createTaskBuilder()
-                .execute(new Runnable() {
-                    @Override
-                    public void run() {
-                        VotifierEvent event = new VotifierEvent(v, Sponge.getCauseStackManager().getCurrentCause());
-                        Sponge.getEventManager().post(event);
-                    }
-                })
-                .async()
```

Comment:
View discussion in PR thread - we need to decide if this is async or sync, and keep it explicit. At the moment, this will call it async within a netty io thread iirc.

Reply:
The point is more the call to the CauseStackManager which will actively throw exceptions if called asynchronously.

Reply:
I understand that ryan, however, the code change also removes the explicit wrapping. I would like that added back in so that we are scheduling it within a sponge scheduler loop, and on top of that doing it explicitly instead of just running it in netty. I agree that the CauseStackManager should be called in sync since it doesn't like not being called in that way, however, we need to be explicit on how we send the event out to the subscribers of the event.

Reply:
There isn't too much of an actual reason to call it asynchronously. I can't see a plugin that couldn't handle the event being fired *synchronously*. I'd expect issues the other way around, to be honest.

Reply:
I would expect issues called async as well - I am very unsure why Tux made the original decision to fire this off async. If we aren't concerned about a possible dependencies on the listeners, then lets wrap this and call it in sync. I'll be sure to add a release note noting this - I don't see issues with people doing weird stuff depending on an async event call, but you never know.

Reply:
The current code is apparently having some alleged issues and failing to receive votes but I’ll test it ASAP and see if that’s true

Reply:
Will this fix the "[Votifier] Unable to process vote from" problem in console. Tried to use Nu in Bungee to have it work across multiple servers.

Reply:
Hopefully!

Reply:
@codeHusky @LoXR I may have fixed the voting issue, netty had an issue where it acted squirrelish with a class we were using from decoding. I would love to PR this as soon as I can so I can get 2.3.6 out the door (probably should be 2.4.0, but whatevs)